core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -110,7 +110,10 @@ trait WorkflowComponent {
     }
 
     def getExecutionServiceIdByExternalId(externalId: String, submissionId: String): ReadAction[Option[String]] = {
-      uniqueResult[String](findWorkflowByExternalIdAndSubmissionId(externalId, UUID.fromString(submissionId)) map (_.executionServiceKey))
+      uniqueResult[WorkflowRecord](findWorkflowByExternalIdAndSubmissionId(externalId, UUID.fromString(submissionId))) map {
+        case Some(wr) => wr.executionServiceKey
+        case _ => None
+      }
```

Comment:
From a quick perusal ... is it possible the `Option[Option[String]]` issue comes from `uniqueResult` and not from a bug in Slick itself? If the column type is `Option[String]`, then `uniqueResult` looks to return `Option[Option[String]]`.

Reply:
Could be the problem is in `ReadAction` instead: that one's mine.